### PR TITLE
HttpURLConnectionLinkValidator

### PR DESCRIPTION
**Pull request by trajano**
Deprecate the OnlineHTTPLinkValidator and used the HttpURLConnection to
utilize what is provided by the JDK including certificates and SSL
updates.

### CODE REVIEW

**Comment by trajano** [comment]
noted in https://issues.apache.org/jira/browse/DOXIATOOLS-55

**Comment by michael-o** [comment]
Please squash commits.

File: doxia-linkcheck/src/main/java/org/apache/maven/doxia/linkcheck/validation/HttpURLConnectionLinkValidator.java [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+            }
+
+            int redirectCount = 0;
+            String content = null;
+            String cookies = null;
+            try
+            {
+                do {
+                    conn = (HttpURLConnection) linkURI.toURL().openConnection();
+                    conn.setRequestProperty( "User-Agent", "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.0)" );
```
Comment (michael-o) [inline doxia-linkcheck/src/main/java/org/apache/maven/doxia/linkcheck/validation/HttpURLConnectionLinkValidator.java]: This should be finally dropped.

**Review by michael-o — APPROVED** [review]

**Comment by michael-o** [comment]
You completely killed proxy support also.

File: doxia-linkcheck/src/main/java/org/apache/maven/doxia/linkcheck/validation/HttpURLConnectionLinkValidator.java (hunk 1 above)
Comment (trajano) [reply doxia-linkcheck/src/main/java/org/apache/maven/doxia/linkcheck/validation/HttpURLConnectionLinkValidator.java]: you mean the user agent thing right?  I'll take it out of the commit then.

**Comment by trajano** [comment]
I'll check how to grab the proxy from the settings.xml there should be only one proxy there anyway.

File: doxia-linkcheck/src/main/java/org/apache/maven/doxia/linkcheck/validation/HttpURLConnectionLinkValidator.java (hunk 1 above)
Comment (michael-o) [reply doxia-linkcheck/src/main/java/org/apache/maven/doxia/linkcheck/validation/HttpURLConnectionLinkValidator.java]: Correct.

**Comment by trajano** [comment]
I'll have to hold off on this one then for the most part my own requirements are satisfied with what this provides.  However, looking through this, I think the better way would be to somehow use `maven-wagon-http` to do the work as that is the closest thing that can be provided by maven framework itself.

I'll make a couple more changes then close off this PR.  One thing I want to avoid is making this thread-unsafe.  So modifying System.properties (which I just removed from the User-Agent)  in a future commit would improve the thread safety.

**Comment by trajano** [comment]
It would've been nice if there was something in plexus that I can use to inject an HTTP client.

**Comment by trajano** [comment]
Closing pull request.  as per my previous comment.

**Comment by michael-o** [comment]
Using Maven Wagon would be perfect here. Just leave it off and pick up when you can.

**Comment by trajano** [comment]
Though personally it should be part of Maven core itself rather than wagon something along the lines of an extra API method that would give us a URL connection based on the settings.xml proxy settings and be thread-safe.  The underlying implementation can be HTTPClient or core Java whatever it better.

**Comment by michael-o** [comment]
I don't think so. Transport isn't Maven core functionality.

**Pull request closed without merging**